### PR TITLE
debug language server startup

### PR DESCRIPTION
**Pull request by friedbrice**
This patch fixes the language server startup logic.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,11 @@ All notable changes to the "alloglot" extension will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 This project adhere's to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [2.0.2] - 2024-02-01
+
+- Debug language server startup.
+- Fix error in example confiuration in README.md
```
Comment (friedbrice) [inline CHANGELOG.md:10]:
```suggestion
- Fix error in example configuration in README.md
```

File: src/client.ts [hunk 2]
```diff
@@ -24,46 +39,27 @@ export function makeClient(config: LanguageConfig): vscode.Disposable {
...
-namespace utils {
-  export function bracket<T, U>(args: { open: () => T, close: (t: T) => void, use: (t: T) => U }): U {
-    let t: T | undefined = undefined
-    try {
-      t = args.open()
-      return args.use(t)
-    }
-    finally {
-      if (t) args.close(t)
-    }
```
Comment (friedbrice) [inline src/client.ts:67]: so it turns out i was closing the server just as soon as i was opening it, i think. actually, i'm not sure what the problem was.

File: src/client.ts [hunk 3]
```diff
@@ -12,6 +12,21 @@ export function makeClient(config: LanguageConfig): vscode.Disposable {
   const { languageId, serverCommand } = config
   if (!languageId || !serverCommand) return vscode.Disposable.from()
 
+  const serverExecutable = {
```
Comment (friedbrice) [inline src/client.ts:15]: The LSP client library will manage the child process for you if you just give it a process spec.

This is how the official Haskell extension launches HLS.

**Review by friedbrice — COMMENTED** [review]
self-reviewed.

**Review by josephsumabat — APPROVED** [review]
LGTM! (with the caveat that my familiarity with the vs-code lsp library is limited)

**Pull request merged**